src/main/java/de/hpi/bpt/argos/api/entityMapping/EntityMappingEndpointImpl.java

```diff
@@ -0,0 +1,288 @@
+
+    private static final Logger logger = LoggerFactory.getLogger(EventTypeEndpointImpl.class);
+    private static final RestEndpointUtil endpointUtil = RestEndpointUtilImpl.getInstance();
+    private static final JsonParser jsonParser = new JsonParser();
+    private static final String EVENT_TYPE_ID_ATTRIBUTE = "EventTypeId";
+    private static final String ENTITY_TYPE_ID_ATTRIBUTE = "EntityTypeId";
+    private static final String MAPPING_CONDITIONS_ATTRIBUTE = "EventEntityMappingConditions";
+    private static final String EVENT_TYPE_ATTRIBUTE_ID_ATTRIBUTE = "EventTypeAttributeId";
+    private static final String ENTITY_TYPE_ATTRIBUTE_ID_ATTRIBUTE = "EntityTypeAttributeId";
+    private static final String TARGET_STATUS_ATTRIBUTE = "TargetStatus";
+
+    private static final String JSON_PARSE_ERROR_MESSAGE = "cannot parse request body to event mapping";
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void setup(Service sparkService) {
+        sparkService.post(EntityMappingEndpoint.getCreateEntityMappingBaseUri(),
+				(Request request, Response response) ->
+						endpointUtil.executeRequest(logger, request, response, this::createEntityMapping));
+
+        sparkService.delete(EntityMappingEndpoint.getDeleteEntityMappingBaseUri(),
+				(Request request, Response response) ->
+						endpointUtil.executeRequest(logger, request, response, this::deleteEntityMapping));
+
+        sparkService.put(EntityMappingEndpoint.getEditEntityMappingBaseUri(),
+				(Request request, Response response) ->
+						endpointUtil.executeRequest(logger, request, response, this::editEntityMapping));
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String createEntityMapping(Request request, Response response) {
+		JsonObject jsonMapping = jsonParser.parse(request.body()).getAsJsonObject();
+
+		if (jsonMapping == null) {
+			halt(HttpStatusCodes.BAD_REQUEST, "no event mapping given in body");
+		}
+
+		createEventMappingFromJson(jsonMapping);
+		return "";
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String deleteEntityMapping(Request request, Response response) {
+        long mappingId = getMappingId(request);
+        EventEntityMapping mapping = PersistenceAdapterImpl.getInstance().getEventEntityMapping(mappingId);
+        if (mapping == null) {
+            halt(HttpStatusCodes.BAD_REQUEST, "given mapping was not found");
+        }
+
+        // delete corresponding mapping conditions
+        List<MappingCondition> conditions = PersistenceAdapterImpl.getInstance().getMappingConditions(mappingId);
+        if (!PersistenceAdapterImpl.getInstance().deleteArtifacts(conditions.toArray(new MappingCondition[conditions.size()]))) {
+            halt(HttpStatusCodes.ERROR, "could not delete mapping conditions");
+        }
+
+        String mappingsUri =  EventTypeEndpoint.getEventTypeEntityMappingsUri(mapping.getEventTypeId());

```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png 'Severity: BLOCKER') NullPointerException might be thrown as 'mapping' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=squid%3AS2259)